yrs/src/doc.rs

```diff
@@ -67,10 +71,102 @@ impl Doc {
         }
     }
 
-    /// Creates a transaction used for all kind of block store operations.
-    /// Transaction cleanups & calling event handles happen when the transaction struct is dropped.
-    pub fn transact(&self) -> Transaction {
-        Transaction::new(self.store.clone())
+    /// Returns a [Text] data structure stored under a given `name`. Text structures are used for
+    /// collaborative text editing: they expose operations to append and remove chunks of text,
+    /// which are free to execute concurrently by multiple peers over remote boundaries.
+    ///
+    /// If not structure under defined `name` existed before, it will be created and returned
+    /// instead.
+    ///
+    /// If a structure under defined `name` already existed, but its type was different it will be
+    /// reinterpreted as a text (in such case a sequence component of complex data type will be
+    /// interpreted as a list of text chunks).
+    pub fn get_text(&self, name: &str) -> Text {
```

Comment:
Now root-level type acquisition happens straight at the document level (without need of transaction). The reason is that building a new y-type requires passing a ref counted pointer to a Store (for purposes of transaction creation), which is available only at the document level or at another y-type level - transaction has only temporary reference to a store, but not a ref counted reference.